Code/Source/RobotImporter/RobotImporterWidget.cpp

```diff
@@ -0,0 +1,106 @@
+/*
+ * Copyright (c) Contributors to the Open 3D Engine Project.
+ * For complete copyright and license terms please see the LICENSE at the root of this distribution.
+ *
+ * SPDX-License-Identifier: Apache-2.0 OR MIT
+ *
+ */
+
+#include <AzCore/IO/Path/Path.h>
+#include <AzCore/Utils/Utils.h>
+#include <QMessageBox>
+#include <QVBoxLayout>
+
+#include "RobotImporter/RobotImporterWidget.h"
+#include "RobotImporter/RobotImporterWidgetUtils.h"
+#include "RobotImporter/URDF/RobotImporter.h"
+
+namespace ROS2
+{
+    RobotImporterWidget::RobotImporterWidget(QWidget* parent)
+        : QWidget(parent)
+        , m_statusLabel("", this)
+        , m_selectFileButton(QObject::tr("Load"), this)
+        , m_importerUpdateTimer(this)
+        , m_robotImporter(
+              [this](RobotImporter::LogLevel level, const AZStd::string& message)
+              {
+                  switch (level)
+                  {
+                  case RobotImporter::LogLevel::Info:
+                      ReportInfo(message);
+                      break;
+                  case RobotImporter::LogLevel::Error:
+                      ReportError(message);
+                      break;
+                  }
+              })
+    {
+        setWindowTitle(QObject::tr("Robot definition file importer"));
+        QVBoxLayout* mainLayout = new QVBoxLayout(this);
+        mainLayout->setSpacing(20);
+        QLabel* captionLabel = new QLabel(QObject::tr("Select a Unified Robot Description Format (URDF) file to import"), this);
+        captionLabel->setWordWrap(true);
+        mainLayout->addWidget(captionLabel);
+        mainLayout->addWidget(&m_selectFileButton);
+        mainLayout->addWidget(&m_statusLabel);
+        mainLayout->addStretch();
+
+        connect(
+            &m_importerUpdateTimer,
+            &QTimer::timeout,
+            [this]
+            {
+                m_robotImporter.CheckIfAssetsWereLoadedAndCreatePrefab(
+                    [this]()
+                    {
+                        m_importerUpdateTimer.stop();
+                        m_selectFileButton.setEnabled(true);
+                    });
+            });
+
+        QObject::connect(
+            &m_selectFileButton,
+            &QPushButton::clicked,
+            this,
+            [this]()
+            {
+                AZStd::optional<AZStd::string> urdfPath = RobotImporterWidgetUtils::QueryUserForURDFPath(this);
+                if (!urdfPath)
+                {
+                    return;
+                }
+
+                AZ::IO::Path prefabName(AZ::IO::PathView(urdfPath.value()).Filename());
+                prefabName.ReplaceExtension("prefab");
+                AZStd::string prefabDefaultPath(AZ::IO::Path(AZ::Utils::GetProjectPath()) / "Assets" / "Importer" / prefabName);
+                auto prefabPath =
+                    RobotImporterWidgetUtils::ValidatePrefabPathExistenceAndQueryUserForNewIfNecessary(prefabDefaultPath, this);
```

Comment:
FYI: you can get an lvalue `AZStd::string&` from an AZ::IO::Path via using the `Native()` method.

That allows interoperability functions that take an AZStd::string or AZStd::fixed_string<1024>